backend/server.js

```diff
@@ -1045,17 +1045,31 @@ app.use(function(req, res, next) {
     }
     
     // add authUID to DB if not already present
-    uCollect.update(
-        {uid: req.authUID},
-        {$setOnInsert: {uid: req.authUID, name: req.authName, dateAdded: (new Date()).toISOString()}},
-        {upsert: true, w: 1},
-        function(err) {
-            if (err) {
-                return sendError(res, 500, "error adding user: " + req.authUID, err);
+    var addAuthUID = function() {
```

Comment:
I think there wouldn't be any harm in unconditionally adding the authUID to the DB. This would simplify the logic flow here as we could just always do this `uCollect.update()` call and then later do the current `userUID !== authUID` check but without the need to call `addAuthUID()`.